plutus-pab/examples/uniswap/Main.hs

```diff
@@ -98,23 +93,18 @@ data UniswapContracts =
 instance Pretty UniswapContracts where
     pretty = viaShow
 
-handleUniswapContract ::
-    ( Member (Error PABError) effs
-    , Member (LogMsg (PABMultiAgentMsg (Builtin UniswapContracts))) effs
-    )
-    => ContractEffect (Builtin UniswapContracts)
-    ~> Eff effs
-handleUniswapContract = Builtin.handleBuiltin getSchema getContract where
-  getSchema = \case
-    UniswapUser _ -> Builtin.endpointsToSchemas @Uniswap.UniswapUserSchema
-    UniswapStart  -> Builtin.endpointsToSchemas @Uniswap.UniswapOwnerSchema
-    Init          -> Builtin.endpointsToSchemas @Empty
-  getContract = \case
-    UniswapUser us -> SomeBuiltin $ Uniswap.userEndpoints us
-    UniswapStart   -> SomeBuiltin Uniswap.ownerEndpoint
-    Init           -> SomeBuiltin US.setupTokens
+instance HasDefinitions UniswapContracts where
+    getDefinitions = [Init, UniswapStart]
+    getSchema = \case
+        UniswapUser _ -> Builtin.endpointsToSchemas @Uniswap.UniswapUserSchema
+        UniswapStart  -> Builtin.endpointsToSchemas @Uniswap.UniswapOwnerSchema
+        Init          -> Builtin.endpointsToSchemas @Empty
+    getContract = \case
+        UniswapUser us -> SomeBuiltin $ Uniswap.userEndpoints us
+        UniswapStart   -> SomeBuiltin Uniswap.ownerEndpoint
+        Init           -> SomeBuiltin US.setupTokens
 
 handlers :: SimulatorEffectHandlers (Builtin UniswapContracts)
```

Comment:
is this still needed?

Reply:
It's used in the `main` of the same file with `Simulator.runSimulationWith handlers $ do ...`